### PR TITLE
Make TPESampler reproducible

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
It seems that currently `TPESampler` is not strictly reproducible because the results of `np.argsort` have randomness by default when there are multiple equal objective values. To make it reproducible, we need to specify `kind="stable"`. 

## Description of the changes
Specify `kind="stable"` to `np.argsort` in `split_observation_pairs`.

### CODE REVIEW

**Comment by c-bata** [comment]
@not522 @HideakiImamura Could you review this PR?

**Comment by contramundum53** [comment]
Note that there may be some performance change introduced by this PR.

**Review by HideakiImamura — COMMENTED** [review]
Could you add a test to validate this change?

**Comment by contramundum53** [comment]
@HideakiImamura I added a reproducibility test for constant objective.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update. LGTM.

**Comment by HideakiImamura** [comment]
@not522 Please merge this PR after your approval.

**Review by not522 — APPROVED** [review]
LGTM

**Pull request merged**
